src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -377,9 +392,98 @@ int fmi3_xml_set_attr_enum(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_
     return 0;
 }
 
-int fmi3_xml_set_attr_boolean(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_t elmID, fmi3_xml_attr_enu_t attrID, int required, unsigned int* field, unsigned int defaultVal) {
+int fmi3_xml_set_attr_boolean(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_t elmID, fmi3_xml_attr_enu_t attrID,
+        int required, unsigned int* field, unsigned int defaultVal)
+{
     jm_name_ID_map_t fmi_boolean_i_dMap[] = {{"true", 1},{"false", 0}, {"1", 1},{"0", 0}, {0,0}};
-    return fmi3_xml_set_attr_enum(context,elmID, attrID,required, field, defaultVal, fmi_boolean_i_dMap);
+    return fmi3_xml_set_attr_enum(context, elmID, attrID, required, field, defaultVal, fmi_boolean_i_dMap);
+}
+
+// TODO: For FMI3, do we want to use bool in the getters for boolean attributes, or keep using unsigned int?
+int fmi3_xml_set_attr_bool(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_t elmID, fmi3_xml_attr_enu_t attrID,
+        int required, bool* field, bool defaultVal)
+{
+    jm_string strVal;
+
+    if (fmi3_xml_get_attr_str(context, elmID, attrID, required, &strVal)) {
+        return -1;
+    }
+
+    if (!strVal && !required) {
+        *field = defaultVal;
+        return 0;
+    }
+    
```

Comment:
`fmi3_xml_get_attr_str` will return false in that case, so wouldn't even get here. It would probably make more sense to remove the check for `required`

Reply:
Can we add this as a comment somehow. It does make sense but makes it hard to follow the code, unless one knows what is happening inside the called functions.

Reply:
Well yes, so I think the confusing part is that we even mention the required attribute. If we just didn't mention it, and it was not used anywhere else, it would be clear that the check was in that other function. But I changed it a bit to make it clear.

Reply:
Great, thanks for clarifying this!